enterprise/views.py

```diff
@@ -859,18 +882,25 @@ def post(self, request):
 
         defer_creation = request.POST.get('defer_creation')
         consent_provided = bool(request.POST.get('data_sharing_consent', False))
-        if defer_creation is None and consent_record.consent_required():
-
+        if defer_creation is None and consent_record.consent_required() and consent_provided:
             # Create EnterpriseCourseEnrollment if we found course_run_id instead of course_key in course_id param.
             # Skip creating EnterpriseCourseEnrollment if we found course_key instead of course_run_id.
 
             # EnterpriseCourseEnrollment will be created when the user will select a course run.
             # A CourseEnrollment record will be created and on the post signal of the CourseEnrollment,
             # an EnterpriseCourseEnrollment record will also get created.
-
             if course_id and self.is_course_run_id(course_id):
                 try:
-                    self.create_enterprise_course_enrollment(request, consent_record, course_id)
+                    self.create_enterprise_course_enrollment(request, consent_record, course_id, license_uuid)
+                    if license_uuid:
+                        enrollment_api_client = EnrollmentApiClient()
+                        course_modes = enrollment_api_client.get_course_modes(course_id)
+                        course_mode = 'verified' if 'verified' in course_modes else 'audit'
```

Comment:
verified isn't the only paid course mode... we should probably check for professional as well.

Reply:
Can a course have both 'verified' and 'professional' modes?

Reply:
i think in practice no, but i doubt there is a formal constraint. if for some reason both exist, we should probably default to 'verified'.

Reply:
I also found the `no-id-professional` course mode. Any input on the `credit` and `honor` course modes?

Reply:
you shouldn't have to worry about either of those, honor is basically the same as audit and credit is complicated, so it's easiest to not deal with it for now.

Reply:
I would use these as the reference point for "modes that allow you to buy stuff": https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py#L171-L172

Reply:
okay, thanks!